@navikt/core/react/src/form/search-field/SearchField.tsx

```diff
@@ -38,50 +24,101 @@ export interface SearchFieldProps
    * SearchField label
    */
   label: React.ReactNode;
+  /**
+   * Inverts color theme
+   * @default false
+   */
+  inverted?: boolean;
+  /**
+   * Customize aria-label on clear button
+   * @default "Slett tekst i felt"
+   */
+  clearButtonLabel?: string;
+  /**
+   * Callback for when user manually clears input with button or Escape
+   */
+  onClear?: () => void;
+  /**
+   * Callback for value in input after change
+   */
+  onChange?: (value: string) => void;
+  /**
+   * Toggles display of "clear"-button when there is text in field
+   */
+  clearButton?: boolean;
 }
 
-interface SearchFieldComponent
-  extends React.ForwardRefExoticComponent<
-    SearchFieldProps & React.RefAttributes<HTMLDivElement>
-  > {
-  Button: SearchFieldButtonType;
-  Clear: SearchFieldClearButtonType;
-  Input: SearchFieldInputType;
-}
-
-const SearchField = forwardRef<HTMLDivElement, SearchFieldProps>(
+const SearchField = forwardRef<HTMLInputElement, SearchFieldProps>(
   (props, ref) => {
-    const {
-      inputProps,
-      errorId,
-      showErrorMsg,
-      hasError,
-      size,
-      inputDescriptionId,
-    } = useFormField(props, "searchfield");
+    const { inputProps, size, inputDescriptionId } = useSearchField(
+      props,
+      "searchfield"
+    );
 
     const {
       className,
       hideLabel,
-      children,
       label,
       description,
-      error,
+      value,
+      clearButtonLabel,
+      onClear,
+      inverted = false,
+      clearButton = true,
       ...rest
     } = props;
 
+    const [controlledValue, setControlledValue] = useState(value ?? "");
+    const searchRef = useRef<HTMLInputElement | null>(null);
+    const mergedRef = mergeRefs([searchRef, ref]);
+    const [wrapperRef, setWrapperRef] = useState<HTMLDivElement | null>(null);
+
+    const handleChange = useCallback(
+      (v: string) => {
+        if (searchRef.current && value === undefined) {
+          searchRef.current.value = v;
+          setControlledValue(v);
+        }
```

Comment:
Denne koden forvirrer meg litt.
1. Handler den egentlig om `clear`-funksjonalitet? (
2. Controlled/uncontrolled-koden er litt enklere andre steder (f.eks. Switch.tsx). Kunne det vært gjort enklere her også?

Reply:
Oppdaterte handleChange og handleClear nå, kan se om det ga mer mening nå.

Trenger en intern controlled value som holder styr på verdien i feltet. Denne blir da brukt til å bestemme om clear-knappen skal vises og om bakgrunnsfargen skal endres for inverted-state

Tanken er at man må sette `searchRef.current.value = "";` når man trykker Escape/onclick på clearbutton. Flyttet denne funksjonaliteten inn i `handleClear` nå da man ikke trenger å gjøre dette i `handleChange`